tests/clients/test_fenda_ditto_client.py

```diff
@@ -0,0 +1,171 @@
+from fl4health.model_bases.fenda_base import FendaModel
+from fl4health.model_bases.sequential_split_models import SequentiallySplitExchangeBaseModel
+from fl4health.parameter_exchange.full_exchanger import FullParameterExchanger
+from tests.clients.fixtures import get_client  # noqa
+from tests.test_utils.models_for_test import FeatureCnn, FendaHeadCnn, HeadCnn
+
+
+@pytest.mark.parametrize("type,model", [(FendaDittoClient, FendaModel(FeatureCnn(), FeatureCnn(), FendaHeadCnn()))])
+def test_setting_initial_weights(get_client: FendaDittoClient) -> None:  # noqa
+    torch.manual_seed(42)
+    fenda_ditto_client = get_client
+    fenda_ditto_client.global_model = SequentiallySplitExchangeBaseModel(FeatureCnn(), HeadCnn())
+    fenda_ditto_client.parameter_exchanger = FullParameterExchanger()
+    config: Config = {"current_server_round": 1}
+
+    params = [val.cpu().numpy() + 1.0 for _, val in fenda_ditto_client.global_model.state_dict().items()]
+    fenda_ditto_client.set_parameters(params, config, fitting_round=True)
+    fenda_ditto_client.update_before_train(1)
+
+    global_model_params = [
+        val.detach().clone()
+        for name, val in fenda_ditto_client.global_model.state_dict().items()
+        if name.startswith("base_module.")
+    ]
+
+    fenda_model_params = [
+        val.detach().clone()
+        for name, val in fenda_ditto_client.model.state_dict().items()
+        if name.startswith("second_feature_extractor.")
+    ]
+
+    # First fitting round we should set both the global and local models to params and store the global model values
+    assert fenda_ditto_client.initial_global_tensors is not None
+    # Tensors should be conv1 weights, biases, conv2 weights, biases, fc1 weights, biases (so 6 total)
+    assert len(fenda_ditto_client.initial_global_tensors) == 4
+    # Make sure that we saved the right parameters
+    for layer_init_global_tensor, layer_params in zip(fenda_ditto_client.initial_global_tensors, params):
+        assert pytest.approx(torch.sum(layer_init_global_tensor - layer_params), abs=0.0001) == 0.0
+    # Make sure the global model was set correctly
+    for global_model_layer_params, layer_params in zip(global_model_params, params):
+        assert pytest.approx(torch.sum(global_model_layer_params - layer_params), abs=0.0001) == 0.0
+    # # Make sure the local model was set correctly
+    for local_model_layer_params, layer_params in zip(fenda_model_params, params):
+        assert pytest.approx(torch.sum(local_model_layer_params - layer_params), abs=0.0001) == 0.0
+
+    # # Now we update in a later round
+    config = {"current_server_round": 2}
+    params = [param + 1.0 for param in params]
+    fenda_ditto_client.set_parameters(params, config, fitting_round=True)
+    fenda_ditto_client.update_before_train(2)
+    global_model_params = [
+        val.detach().clone()
+        for name, val in fenda_ditto_client.global_model.state_dict().items()
+        if name.startswith("base_module.")
```

Comment:
With a full parameter exchanger, we should match the params for all components of the `global_model_params` not just those in the base_module, if I'm not mistaken?

Reply:
yes! removing the `if...`